### PR TITLE
Backported forgotten vanilla/wotlk changes

### PR DESCRIPTION
**Pull request by danielsreichenbach**
## 🍰 Pullrequest
Just some small changes from vanilla/wotlk branches that seem to be lost in merging. This basically brings framework library to equal state in classic/tbc/wotlk and syncs up shared library as much as possible.

### Proof
- None

### Issues
- None

### How2Test
- None

### Todo / Checklist
- [X] None

### CODE REVIEW

**Comment by killerwife** [comment]
Fine with me, but I will wait on all PRs what the old man @cyberium says.

File: src/shared/Metric/Measurement.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ std::stringstream& operator<<(std::stringstream& out, const Measurement& measure
         else if (field.second.type() == typeid(float))
             out << boost::any_cast<float>(field.second);
         else if (field.second.type() == typeid(bool))
-            out << (boost::any_cast<bool>(field.second) ? "t" : "f");
```
Comment (Laizerox) [inline src/shared/Metric/Measurement.cpp:60]: This line should be actually backported to other cores instead. See this commit 03f6a35ce3eb524f81ce84e34f2e948f41ce060d for more information.

Comment (danielsreichenbach) [reply src/shared/Metric/Measurement.cpp]: Ah sorry, missed that. Reverted and put the proper version into vanilla and wotlk PRs.

**Review by cyberium — APPROVED** [review]

**Comment by killerwife** [comment]
Pr much appreciated mr TL

**Pull request merged**